### ISSUE TITLE
Regression in v0.32.0: flip operation can cause "out of order read" error

### ISSUE THREAD

**Issue by jwerre** [lovell/sharp#3600]
## Possible bug

I've run into and issue where when I try to flip an image I get the following error:

```
Error: pngload_buffer: out of order read at line 1102
```

I've tried other transformations (including `flop`) on the same image without issues.

### Is this a possible bug in a feature of sharp, unrelated to installation?

- [X] Running `npm install sharp` completes without error.
- [X] Running `node -e "require('sharp')"` completes without error.

If you cannot confirm both of these, please open an [installation issue](https://github.com/lovell/sharp/issues/new?labels=installation&template=installation.md) instead.

### Are you using the latest version of sharp?

- [X] I am using the latest version of `sharp` as reported by `npm view sharp dist-tags.latest`.

npm view sharp dist-tags.latest
0.32.0

### What is the output of running `npx envinfo --binaries --system --npmPackages=sharp --npmGlobalPackages=sharp`?

Ok to proceed? (y)

  System:
    OS: Linux 5.15 Debian GNU/Linux 10 (buster) 10 (buster)
    CPU: (5) arm64 unknown
    Memory: 9.80 GB / 11.68 GB
    Container: Yes
    Shell: 5.0.3 - /bin/bash
  Binaries:
    Node: 16.18.1 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 8.19.2 - /usr/local/bin/npm

I'm running this inside the node:16.18-slim Docker container

### What are the steps to reproduce?

```js
const processImage = async function (buffer, options = {}) {

	const image = sharp(buffer);

	for (const key of Object.keys(options)) {
		const val = options[key];
		if (val && typeof image[key] === 'function') {
			image[key](val);
		}
	}

	let result;

	try {
		result = await image.toBuffer();
	} catch (err) {
		console.log(err); // Error: pngload_buffer: out of order read at line 1102
		throw err;
	}

	return result;
	
};


const file = readFileSync('./test/assets/test_img.png');
const floppedBuff = await processImage(file, {flop: true}); // OK
const flippedBuff = await processImage(file, {flip: true}); // ERROR
```

### What is the expected behaviour?

Flipped image should not produce an error.

### Please provide a minimal, standalone code sample, without other dependencies, that demonstrates this problem

See above.

### Please provide sample image(s) that help explain this problem

![test_img](https://user-images.githubusercontent.com/747587/227634355-6900799a-cb3e-41a5-90dc-681fec64ae9e.png)

**Comment by lovell** [comment]
Thanks for reporting, this will be a side effect of the switch in the newly-released v0.32.0 to more eagerly taking advantage of libvips' sequential read feature. We need to add `flip` to the list of operations that will force the pipeline as a whole to use random access read instead.

https://github.com/lovell/sharp/blob/b9d4c30a9fe203672259a0ef99d3551e6e456114/src/pipeline.cc#L1655-L1670

For this sample image, it looks like the PNG decoder simply can't deliver pixel values quickly enough to libvips' very fast flip operation, whereas the flop operation is slower so it still works.

You can work around it for now with:
```js
sharp(input, { sequentialRead: false }).flip()...
```

**Comment by lovell** [comment]
v0.32.1 now available.
